src/components/DateTimePicker/TimePicker.tsx

```diff
@@ -52,22 +58,93 @@ export class TimePicker extends React.Component<ITimePickerProps, void> {
         this.props.onTimeChanged(this.props.hour, value);
     }
 
+    normalizeHours(hour: number, is24Hour: boolean, isPeriodPm: boolean) {
+        if (!is24Hour) {
+            if (hour > 12) {
+                hour = 12;
+            }
+            if (isPeriodPm) {
+                if (hour < 12) {
+                    hour += 12;
+                }
+            } else {
+                if (hour === 12) {
+                    hour = 0;

```

Comment:
The **if hour > 12** line clips the value because input went over the allowed max value, so it's being handled here, setting it to max value for 12-hour format. After that, conversion to 24-hour format follows. If the period is PM, 12 hours should be added to the input value (except when it is 12 PM), and if AM, 12 AM will be converted to 0.  
Function always converts to [0-24], but starts with correcting input errors.